build/parseList.c

```diff
@@ -8,13 +8,99 @@
 #include "build/rpmbuild_internal.h"
 #include "debug.h"
 
+static int addLinesFromFile(rpmSpec spec, const char * const fn, rpmTagVal tag)
+{
+    int nlines = 0;
+    int flags = STRIP_COMMENTS | STRIP_TRAILINGSPACE;
+    ARGV_t argv = NULL;
+
+    int terminate = rpmExpandNumeric(tag == RPMTAG_SOURCE
+				     ? "%{?_empty_sourcelist_terminate_build}"
+				     : "%{?_empyy_patchlist_terminate_build}");
```

Comment:
The patchlist macro has a typo (empyy vs empty). However the bigger thing is that I don't think these should be configurable. With %files, an empty %files section is special as that controls whether the package gets built at all, but for source and patch lists, I don't see any reason to allow them to be empty.

Reply:
I would argue the opposite: Why would they not be allowed to be empty. Especially the patches file. Think of the patch list being generated automatically from a git repository. After a rebase the patch list may be empty. Why require the spec file to be adjusted in this case instead of just working?
Sources may be a bit different as one would expect that the packagers have a tighter look on them. But one could still imagine some add-on stuff like docs or test scripts that could be added dynamically and might or might not be there.

Reply:
Fair enough I guess. But if we consider it valid then let it always be valid. We have way, way too many of these fiddly configurables as it is.

Reply:
Aaaaand is gone!